src/main/java/net/minecraftforge/client/LevelRendererHooks.java

```diff
@@ -0,0 +1,118 @@
+package net.minecraftforge.client;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.math.Matrix4f;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.LevelRenderer;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.resources.ResourceLocation;
+
+/**
+ * Renders registered hooks to render during their appropriate {@link Phase}.
+ * Renderer hooks may be registered at any time.
+ * 
+ * @see #register
+ */
+public class LevelRendererHooks
+{
+    private static final HashMap<Phase, List<Consumer<RenderContext>>> HOOKS = new HashMap<>();
+    private static final HashMap<RenderType, Phase> RENDER_TYPE_PHASES = new HashMap<>();
+    public static float partialTicks = 0.0F;
+
+    /**
+     * Registers the Consumer passed to the {@link Phase} specified.
+     */
+    public static void register(Phase phase, Consumer<RenderContext> hook)
+    {
+        HOOKS.computeIfAbsent(phase, p -> new ArrayList<>()).add(hook);
+    }
+
+    /**
+     * Renders registered hooks for the {@link Phase} passed.
+     */
+    public static void render(Phase phase, LevelRenderer levelRenderer, PoseStack poseStack, Matrix4f projectionMatrix, int ticks, double camX, double camY, double camZ)
+    {
+        List<Consumer<RenderContext>> hooks = HOOKS.get(phase);
+        if (hooks != null)
+        {
+            Minecraft.getInstance().getProfiler().popPush(phase.toString());
+            for (Consumer<RenderContext> hook : hooks)
+                hook.accept(new RenderContext(levelRenderer, poseStack, projectionMatrix, ticks, partialTicks, camX, camY, camZ));
```

Comment:
It would be better to create the context once before the loop and pass that object in instead of creating it for every hook. This is of course assuming that `RenderContext` is an immutable structure, which seems like it is.